rero_ils/modules/notifications/subclasses/request.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019 RERO
+# Copyright (C) 2020 UCLouvain
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""API for manipulating "request" circulation notifications."""
+
+from __future__ import absolute_import, print_function
+
+from rero_ils.filter import format_date_filter
+from rero_ils.modules.documents.dumpers import DocumentNotificationDumper
+from rero_ils.modules.items.dumpers import ItemNotificationDumper
+from rero_ils.modules.loans.api import LoanState
+from rero_ils.modules.patrons.dumpers import PatronNotificationDumper
+from rero_ils.utils import language_iso639_2to1
+
+from .internal import InternalCirculationNotification
+
+
+class RequestCirculationNotification(InternalCirculationNotification):
+    """Request circulation notifications class.
+
+    A request notification is a message send to a library to notify that a
+    document has been requested by a patron. As it's a internal notification,
+    it should never be cancelled (except if the requested item doesn't exist
+    anymore) and are always send by email.
+
+    Request notification works synchronously. This means it will be send just
+    after the creation. This also means that it should never be aggregated.
+    """
+
+    def get_recipients(self):
+        """Get notification recipient email addresses."""
+        # Request notification will be sent to the loan location if it define
+        # a ``notification_email``, otherwise to the library address.
```

Comment:
```suggestion
        # Request notification will be sent to the item location if 
        # a location ``notification_email`` is defined, otherwise to the library address.
```

If I got it right.

Reply:
👍